src/actions/articleActions.js

```diff
@@ -12,12 +12,50 @@ export const fetchArticlesFailure = errorMessage => ({
   errorMessage,
 });
 
-const fetchArticlesThunk = () => dispatch => axios.get(`${APP_URL}/articles`)
+export const showErrorAction = payload => ({
+  type: ACTION_TYPE.SHOW_ERROR,
+  payload, // error message
+});
+
+export const fetchArticlesThunk = () => dispatch => axios.get(`${APP_URL}/articles`)
   .then((response) => {
     dispatch(fetchArticlesSuccess(response.data.results));
   })
   .catch(() => {
     dispatch(fetchArticlesFailure('Check your internet conectivity'));
   });
 
+
+export const getArticleAction = payload => ({
+  type: ACTION_TYPE.GET_ARTICLE,
+  payload, // entire article
+});
+
+export const getLikeStatusAction = payload => ({
+  type: ACTION_TYPE.GET_LIKE_STATUS,
+  payload,
+});
+
+export const getArticleThunk = articleId => (dispatch) => {
+  const url = `https://ah-backend-thanos-staging.herokuapp.com/api/articles/${articleId}`;
+  return axios.get(url)
+    .then((response) => {
+      dispatch(getArticleAction(response.data.results));
+      return response;
+    })
+    .catch((error) => {
+      dispatch(showErrorAction({ ...alert, text: error.response ? error.response.data.results.error : 'Article not found' }));
+    });
+};
+
+export const getLikeStatusThunk = ({ articleId, token }) => (dispatch, getState) => {
+  const url = `https://ah-backend-thanos-staging.herokuapp.com/api/articles/${articleId}/like_status`;
```

Comment:
replace the url with the APP_URL from the utils folder

Reply:
Alright